File: mobikulOC/src/main/java/webkul/opencart/mobikul/model/ProductCategory/Category.java
package webkul.opencart.mobikul.model.ProductCategory;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by manish.choudhary on 28/7/17.
 */

public class Category implements Parcelable {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("path")
    @Expose
    private String path;

    /**
     * No args constructor for use in serialization
     *
     */
    public Category() {
    }

    /**
     *
     * @param name
     * @param path
     */
    public Category(String name, String path) {
        super();
        this.name = name;
        this.path = path;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }


    protected Category(Parcel in) {
        name = in.readString();
        path = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(path);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<Category> CREATOR = new Parcelable.Creator<Category>() {
        @Override
        public Category createFromParcel(Parcel in) {
            return new Category(in);
        }

        @Override
        public Category[] newArray(int size) {
            return new Category[size];
        }
    };
}

File: mobikulOC/src/main/java/webkul/opencart/mobikul/model/GetHomePage/Category.kt
package webkul.opencart.mobikul.model.GetHomePage

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Category {
    @SerializedName("child_status")
    @Expose
    var childStatus: Boolean? = null
    @SerializedName("dominant_color")
    @Expose
    var dominantColor: String? = null
    @SerializedName("dominant_color_icon")
    @Expose
    var dominantColorIcon: String? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("children")
    @Expose
    var children: List<Child>? = null
    @SerializedName("column")
    @Expose
    var column: String? = null
    @SerializedName("path")
    @Expose
    var path: String? = null
    @SerializedName("image")
    @Expose
    var image: String? = null
    @SerializedName("icon")
    @Expose
    var icon: String? = null
    @SerializedName("thumb")
    @Expose
    var thumb: String? = null

}